charts/consul/templates/gateway-resources-configmap.yaml

```diff
@@ -17,6 +17,7 @@ data:
     {{ toJson .Values.connectInject.apiGateway.managedGatewayClass.resources }}
   {{- end }}
   {{- if (mustHas "resource-apis" .Values.global.experiments) }}
+  {{- if .Values.meshGateway.enabled }}
```

Comment:
should we just `and` these two statements rather than two `if` statements?

Reply:
+1

Reply:
😆 I considered that, but the one combined `if` winds up being harder to digest IMHO. If it's something you feel strongly about, I'm down to change it

Reply:
I don't feel particularly strongly either way tbh, not combining them might make it easier to find all places where the experiments check is used and remove them when we need to do that later

Reply:
Here's my +1 for using `and` for both statements, .`Values.meshGateway.enabled` I'm assuming based on my manual tests cannot be used to enable v2 without `  experiments: - "resource-apis"` added to the yaml, so I think we should add the check here to prevent runtime errors

Reply:
It's slightly different: `experiments: [resource-apis]` enables v2. `meshGateway.enabled` can be set in v1 or v2